lib/kube/proxy/pod_rbac_test.go

```diff
@@ -0,0 +1,539 @@
+					ErrStatus: metav1.Status{
+						Status:  "Failure",
+						Message: "[00] access denied",
+						Code:    403,
+					},
+				},
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			// generate a kube client with user certs for auth
+			client, _ := testCtx.genTestKubeClientTLSCert(
+				t,
+				tt.args.user.GetName(),
+				kubeCluster,
+			)
+
+			rsp, err := client.CoreV1().Pods(tt.args.namespace).List(
+				testCtx.ctx,
+				metav1.ListOptions{},
+			)
+			require.NoError(t, err)
+
+			require.Equal(t, tt.want.listPodsResult, getPodsFromPodList(rsp.Items))
+
+			_, err = client.CoreV1().Pods(metav1.NamespaceDefault).Get(
+				testCtx.ctx,
+				testPodName,
+				metav1.GetOptions{},
+			)
+			require.Equal(t, tt.want.getTestPodResult, err)
+		})
+	}
+}
+
+func getPodsFromPodList(items []corev1.Pod) []string {
+	pods := make([]string, 0, len(items))
+	for _, item := range items {
+		pods = append(pods, filepath.Join(item.Namespace, item.Name))
+	}
+	return pods
+}
+
+func TestWatcherResponseWriter(t *testing.T) {
+	t.Parallel()
+	fakeEvents := fakeEvents()
+	log := logrus.New()
+	log.SetLevel(logrus.DebugLevel)
+	type args struct {
+		allowed []types.KubernetesResource
+		denied  []types.KubernetesResource
+	}
+	tests := []struct {
+		name string
+		args args
+		want []*metav1.WatchEvent
+	}{
+		{
+			name: "receive every event",
+			args: args{
+				allowed: []types.KubernetesResource{
+					{
+						Kind:      types.KindKubePod,
+						Namespace: "*",
+						Name:      "*",
+					},
+				},
+			},
+			want: fakeEvents,
+		},
+		{
+			name: "receive events for default namespace",
+			args: args{
+				allowed: []types.KubernetesResource{
+					{
+						Kind:      types.KindKubePod,
+						Namespace: "default",
+						Name:      "*",
+					},
+				},
+			},
+			want: fakeEvents[1:],
+		},
+		{
+			name: "receive events for default namespace but with denied pod",
+			args: args{
+				allowed: []types.KubernetesResource{
+					{
+						Kind:      types.KindKubePod,
+						Namespace: "default",
+						Name:      "*",
+					},
+				},
+				denied: []types.KubernetesResource{
+					{
+						Kind:      types.KindKubePod,
+						Namespace: "default",
+						Name:      "otherPod",
+					},
+				},
+			},
+			want: fakeEvents[1:2],
+		},
+		{
+			name: "receive receives no events for default namespace",
+			args: args{
+				allowed: []types.KubernetesResource{
+					{
+						Kind:      types.KindKubePod,
+						Namespace: "default",
+						Name:      "rand*",
+					},
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			userReader, userWriter := io.Pipe()
+			negotiator := newClientNegotiator()
+			filterWrapper := newPodFiltererBuilder(tt.args.allowed, tt.args.denied, log)
+			watcher, err := responsewriters.NewWatcherResponseWriter(newFakeResponseWriter(userWriter), negotiator, filterWrapper)
```

Comment:
```suggestion
			watcher, err := responsewriters.NewWatcherResponseWriter(newFakeResponseWriter(userWriter), negotiator, filterWrapper)
			require.NoError(t, err)
```

and drop the require.NoError below?